### ISSUE TITLE
FromCache fails for queries with filter_by()

### ISSUE THREAD

**Issue by prymitive** [iurisilvio/Flask-SQLAlchemy-Cache#3]
It works fine for ``Model.query.options(FromCache(cache)).all()`` but fails once I want to filter results:

```python
Model.query.options(FromCache(cache)).filter_by(active=True).all()
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2320, in all
    return list(self)
TypeError: iter() returned non-iterator of type 'list'
```

**Comment by iurisilvio** [comment]
I wasn't able to reproduce this issue. Can you provide a small broken example?

**Comment by prymitive** [comment]
Sure, I'll try to create a dummy app, it might be specific to my setup. I'll get back tomorrow, thanks

**Comment by prymitive** [comment]
I've refactored my code and the issue have gone away, can't reproduce it any longer. Please ignore me, that was some glitch on my side, thanks

**Comment by iurisilvio** [comment]
No problem. Thanks for your feedback!

**Comment by prymitive** [comment]
it seems that there might be a issue after all, but it happens when the query doesn't match anything, there is nothing in the db or the value is not cached, not sure yet, give me a minute to figure it out

**Comment by iurisilvio** [comment]
Thanks! I reproduced it here. It happens with zero results.

**Comment by prymitive** [comment]
It seems that in case of zero results __iter__ method would return an empty list object instead of iterator instance, the fix from #4 might be enough
